File: PAT (Basic-Level)/C/1007.c
#include <stdio.h>
int main()
{
    int n,p1,p2,d,count,i,j,s[100000];
    count=0;
    p1=1;
    p2=3;
    scanf("%d",&n);
    for(i = 2;i <= n;i++){
        s[i] = 1;
    }
    for(i = 2;i * i <= n;i++){
        if(s[i] == 1){
            for(j = i;i*j <= n;j++){
            s[i*j] = 0;
            }
        }
    }
    for(i=5;i<=n;i++){
        if(s[i]==1){
            if((i-p2)==2){
                count++;
            }
            p1=p2;
            p2=i;
        }
    }
    printf("%d\n",count);
    return 0;
}

File: PAT (Basic-Level)/Python/1007.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

import math  

def isPrime(N):  # judge whether
    for i in range(2, int(math.sqrt(N))+1, 1):  
        if N % i == 0:  
            return False  
    return True  

n = int(input())

'''count = 0
for i in range(3, n-1): # range(1, n) means from 1 to n-1
    if isPrime(i) and isPrime(i + 2):
        count = count + 1
print(count)
#overtime!!!
'''
#do not know why
result = [i for i in range(3, n-1) if isPrime(i) and isPrime(i+2)]  # put the suitable i in the list li
print(len(result)) 
